### ISSUE TITLE
Fluid compiler warnings

### ISSUE THREAD

**Issue by Albrecht-S** [fltk/fltk#622]
```
[1/2] /usr/bin/c++  ...  -Wall -Wunused -Wsuggest-override -g -std=c++2a ... /Fluid_Image.cxx.o -MF fluid/CMakeFiles/fluid.dir/Fluid_Image.cxx.o.d -o fluid/CMakeFiles/fluid.dir/Fluid_Image.cxx.o -c ../../fluid/Fluid_Image.cxx
../../fluid/Fluid_Image.cxx: In member function ‘void Fluid_Image::write_static(int)’:
../../fluid/Fluid_Image.cxx:141:15: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘int’ [-Wsign-compare]
  141 |     if (nData == -1) write_file_error("GIF");
      |         ~~~~~~^~~~~
../../fluid/Fluid_Image.cxx:153:15: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘int’ [-Wsign-compare]
  153 |     if (nData == -1) write_file_error("BMP");
      |         ~~~~~~^~~~~
../../fluid/Fluid_Image.cxx:206:15: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘int’ [-Wsign-compare]
  206 |     if (nData == -1) write_file_error("JPEG");
      |         ~~~~~~^~~~~
../../fluid/Fluid_Image.cxx:218:15: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘int’ [-Wsign-compare]
  218 |     if (nData == -1) write_file_error("PNG");
      |         ~~~~~~^~~~~
../../fluid/Fluid_Image.cxx:233:19: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘int’ [-Wsign-compare]
  233 |         if (nData == -1) write_file_error("SVGZ");
      |             ~~~~~~^~~~~
../../fluid/Fluid_Image.cxx:239:19: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘int’ [-Wsign-compare]
  239 |         if (nData == -1) write_file_error("SVG");
      |             ~~~~~~^~~~~
../../fluid/Fluid_Image.cxx: At global scope:
../../fluid/Fluid_Image.cxx:54:12: warning: ‘bmp_header_written’ defined but not used [-Wunused-variable]
   54 | static int bmp_header_written = 0;
      |            ^~~~~~~~~~~~~~~~~~
```

These are actually two different types of warnings, one repeated several times, the other one (`[-Wunused-variable]`) appears only once.
Notes:
- commandline edited
- Git master
- commit bfe730ecbf94fb709928b735cf52c41e1e4d7ad3
- I suggest to compile/build with `-Wall -Wunused -Wsuggest-override` to find such warnings during development

FWIW, here is a part of my setup script which I'm using whenever I create/use a new branch. YMMV.
```
C_FLAGS_INIT="-Wall -Wunused"
CXX_FLAGS_INIT="-Wall -Wunused -Wsuggest-override"

# CXX_STANDARD:   -DCMAKE_CXX_STANDARD=20    # valid: 98, 11, 14, 17, 20, 23, 26
#                 https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html

CXX_STANDARD="-DCMAKE_CXX_STANDARD=20 -DCMAKE_CXX_EXTENSIONS=OFF"

cmake -G Ninja -DCMAKE_BUILD_TYPE=Debug \
  -DCMAKE_C_FLAGS_INIT="$C_FLAGS_INIT" \
  -DCMAKE_CXX_FLAGS_INIT="$CXX_FLAGS_INIT" $CXX_STANDARD \
  ../..
```

HTH.

**Comment by MatthiasWM** [comment]
Thanks. I will fix that tomorrow.

**Comment by MatthiasWM** [comment]
Should be fixed in 9a3f0f2089a98c3705d3f3e58aa235b808d51b8f and ab0d59220eb341daea4ac30bb4b506370de98eb0

**Comment by MatthiasWM** [comment]
The second one was actually a real error.
